web/vibe/web/rest.d

```diff
@@ -1987,6 +2068,159 @@ private string generateModuleImports(I)()
 	return join(map!(a => "static import " ~ a ~ ";")(modules), "\n");
 }
 
+/***************************************************************************
+
+	The client sends the list of allowed content types in the 'Allow' http header
+	and the response will contain a 'Content-Type' header. This function will
+	try to find the best matching @SerializationResult UDA based on the allowed
+	content types.
+
+	Note:
+
+	Comment 1: if the request doesn't specify any allowed content types, then * / *
+	is assumed
+
+	Comment 2: if there are no UDA's matching the client's allowed content types, -1
+	is returned
+
+	Comment 3: if there are more than 1 matching UDA, for ONE specific client's allowed
+	content type(and their priority is the same - see below),
+	then the one specified earlier in the code gets chosen
+
+	Comment 4: accept-params(quality factor) and accept-extensions are ignored
+	https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
+
+	Comment 5: matching the most specific content type without any wildcard has priority
+
+	Comment 6: the request's content type can be in the format of
+
+	$(UL
+		$(LI major type / minor type)
+		$(LI major type / *)
+		$(LI * / *)
+	)
+
+	Params:
+		T = compile time known ResultSerializer classes
+		req_content_types_str = list of allowed content types for example:
+		text/*;q=0.3, text/html;q=0.7, text/html;level=1
+
+	Returns:
+		index of the result serializers in the T... AliasSeq if matching found,
+		-1 otherwise
+
+***************************************************************************/
+
+package int get_matching_content_type (T...)(string req_content_types_str) pure @safe
```

Comment:
This whole `Content-Type` handling stuff would be better suited in `vibw.inet`, wouldn't it @s-ludwig ?
E.g. https://github.com/vibe-d/vibe.d/blob/master/inet/vibe/inet/mimetypes.d

Reply:
Of course, it would mean to be made more generic, to allow matching content types with a list of "allow".

Reply:
It would indeed fit there better, but in general I'm okay with keeping such `private`/`package` things locally if their API is not yet suitable for a public interface. Otherwise it increases the PR review complexity and time-to-merge, which usually is the biggest issue for overall progress.

But this is definitely something to keep in mind/as a follow-up ticket.